src/client/add-metainfo-releases

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/python3
+
+import argparse
+import re
+import shlex
+import subprocess
+import time
+import xml.etree.ElementTree as ET
+
+
+# Pretend like this is a little bit functional
+def element(tag, text=None, children=(), **kwargs):
+    tag = ET.Element(tag, kwargs)
+    tag.text = text
+    tag.extend(children)
+    return tag
+
+
+def points_from_body(body):
+    for text in re.split('^ *-', body, flags=re.MULTILINE):
+        if point := ' '.join(text.split()).strip():
+            yield element('li', point)
+
+
+def format_release(blob):
+    version, date, body = blob.split('|', 2)
+    url = element('url', f'https://cockpit-project.org/blog/cockpit-{version}.html', type='details')
+    ul = element('ul', children=points_from_body(body))
+    description = element('description', children=(ul,))
+    return element('release', version=version, date=date, children=(url, description))
+
+
+def get_releases():
+    format = '%(if)%(taggerdate)%(then)%(refname:short)|%(taggerdate:short)|%(contents:body)%(end)'
+    output = subprocess.check_output(['git', 'for-each-ref', '--shell', '--sort=-taggerdate',
+                                      '--contains', '258', '--format', format, 'refs/tags'], text=True)
+    return shlex.split(output)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--version', help='Current version number', default='')
+    parser.add_argument('xml', help='Path to xml.in file in tree')
+    args = parser.parse_args()
+
+    tree = ET.parse(args.xml)
+    releases = tree.find('releases')
+
+    # Only update the file if it doesn't already contain releases.
+    if not releases.find('release'):
+        if 'g' in args.version:
+            today = time.strftime('%Y-%m-%d')
+            releases.append(element('release', version=args.version, date=today, type='development'))
+        releases.extend(format_release(r) for r in get_releases())
+        ET.indent(releases, space="  ", level=1)
```

Comment:
This is not present in Ubuntu 20.04 yet, see [failed workflow](https://github.com/cockpit-project/cockpit/runs/5773904226?check_suite_focus=true). As 20.04 LTS is a very popular platform for all kinds of infrastructure, let's find an API that works there. I'm sure ElementTree has *some* pretty-printer?